python/cuspatial/cuspatial/core/gis.py

```diff
@@ -163,16 +172,36 @@ def point_in_polygon_bitmap(
     DataFrame: a DataFrame of Boolean values indicating whether each point
     falls within each polygon.
     """
-    bitmap_result = cpp_point_in_polygon_bitmap(
-        x_points, y_points, polygon_fpos, polygon_rpos, polygons_x, polygons_y
+
+    if len(poly_offsets) == 0:
+        return DataFrame()
+
+    (
+        test_points_x,
+        test_points_y,
+        poly_points_x,
+        poly_points_y,
+    ) = normalize_point_columns(
+        as_column(test_points_x),
+        as_column(test_points_y),
+        as_column(poly_points_x),
+        as_column(poly_points_y),
+    )
+
+    result = cpp_point_in_polygon(
+        test_points_x,
+        test_points_y,
+        as_column(poly_offsets, dtype="int32"),
+        as_column(poly_ring_offsets, dtype="int32"),
+        poly_points_x,
+        poly_points_y,
     )
 
-    result_binary = gis_utils.pip_bitmap_column_to_binary_array(
-        polygon_bitmap_column=bitmap_result, width=len(polygon_fpos)
+    result = gis_utils.pip_bitmap_column_to_binary_array(
```

Comment:
Wow, so we are exploding the bitmaps to boolean columns anyway... So it probably would be more efficient (and eliminate the 32-poly limit) to just do that in C++...

Reply:
I think this to cast the matrix columns to `dtype="bool"` just below, we could definitely do this without numba.